spyder/app/tests/test_mainwindow.py

```diff
@@ -3221,5 +3221,33 @@ def test_run_unsaved_file_multiprocessing(main_window, qtbot):
             lambda: 'None' in shell._control.toPlainText())
 
 
+@pytest.mark.slow
+@flaky(max_runs=3)
```

Comment:
Why is the flaky needed?

Did it fail intermittently when testing this PR?

Reply:
Sometimes the kernel fails to start or restart during the prescribed timeout, so we decided to mark these tests as flaky to avoid a full run of our test suite.